expyfun/_experiment_controller.py

```diff
@@ -1151,6 +1156,8 @@ def text_input(self, stop_key='return', pos=[0, 0], color='white',
                              font_name=font_name, font_size=font_size,
                              wrap=wrap, units=units, log_data=False)
             self.flip()
+        self.call_on_next_flip(partial(self.write_data_line, 'text_input',
+                                       text))
```

Comment:
Why wait until `call_on_next_flip` rather than immediately do it (i.e., call `self.write_data_line(...)` directly)?

Reply:
I copied from ``ec.screen_text()`` assuming there was a reason I didn't know about. Doing it directly makes more sense to me.

Reply:
In the case of `ec.screen_text` the text does not actually get displayed until the next call to `.flip()`. Here the text has already been obtained/completed